src/ranking.jl

```diff
@@ -6,50 +6,63 @@
 # The implementations here follow this wikipedia page.
 #
 
-
 function _check_randparams(rks, x, p)
     n = length(rks)
     length(x) == length(p) == n || raise_dimerror()
     return n
 end
 
+# ranking helper function: calls sortperm(x) and then ranking method f!
+function _rank(f!, x::AbstractArray, R::Type=Int; sortkwargs...)
+    rks = similar(x, R)
+    ord = reshape(sortperm(vec(x); sortkwargs...), size(x))
+    return f!(rks, x, ord)
+end
 
+# ranking helper function for arrays with missing values
+function _rank(f!, x::AbstractArray{>: Missing}, R::Type=Int; sortkwargs...)
+    inds = findall(!ismissing, vec(x))
+    isempty(inds) && return missings(R, size(x))
+    T = nonmissingtype(eltype(x))
+    xv = Vector{T}(undef, length(inds))
+    @inbounds for (i, ind) in enumerate(inds)
+        xv[i] = x[ind]
+    end
```

Comment:
Why don't we just do `xv = view(x, inds)`? If making a copy is faster, then `xv = disallowmissing(view(x, inds))` as in the old code or `xv = convert(AbstractArray{T}, x)` is both shorter and much more efficient e.g. for `CategoricalArray`.